public/ticketmanager.php

```diff
@@ -215,17 +232,31 @@
             sanitize_outputs(
                 $assignedToUser,
             );
+            $reportedByUser = requestInputSanitized('p', null);
+            if (!isValidUsername($reportedByUser)) {
+                $reportedByUser = null;
+            }
+            sanitize_outputs(
+                $reportedByUser,
+            );
+            $resolvedByUser = requestInputSanitized('r', null);
+            if (!isValidUsername($resolvedByUser)) {
+                $resolvedByUser = null;
+            }
+            sanitize_outputs(
+                $resolvedByUser,
+            );
```

Comment:
`sanitize_outputs` should only be used for things that are getting written to the page. These are going to be passed to `countOpenTickets` in a few lines where they'll be used in queries. That should be calling `sanitize_inputs` on them, and won't expect them to have already been escaped for HTML.

Reply:
$assignmedToUser, $resolvedByUser and $reportedByUser actually do get written out after selecting a developer for any of them to notate what value it currently has. So I think in that case it is safe to keep these sanitize_output calls correct?
![image](https://user-images.githubusercontent.com/4047771/164111993-b569ba5f-5768-40ed-873d-88b3f9b694df.png)

I did clean up the text boxes though which were also displaying the value which I didn't think of until now.

Reply:
My point is that you're escaping the value before using it in a query. I think `sanitize_outputs` only escapes `<` and `>` because of the flags that we pass to `htmlentities`, but if those were valid username characters, and you tried to match `User<DEV>`, `$resolvedByUser` would be `User&lt;DEV&gt;` going into `countOpenTickets`, and wouldn't match the `User<DEV>` in the database.

You should either sanitize them later (after they've been used in the query), or create a copy to sanitize.

Reply:
Oh wow yeah I misunderstood what you were saying and what I was doing with those, I believe I fixed this now by moving it down to where the html starts building and also cleaned up the sanitize_outputs call to combine the three variables into one call since it supports that.